airbnb_project/listings/views.py

```diff
@@ -11,29 +11,39 @@
 # Set up logger for this module
 logger = logging.getLogger(__name__)
 
+# Global flag to track if the spider is currently running
+is_scraping = False
 
-def run_spider_in_thread(runner):
-    """
-    Run the Scrapy spider in a separate thread.
 
-    This function is designed to be run in a separate thread to avoid
-    blocking the main Django process. It handles the Scrapy crawling
-    process and manages the reactor.
+def handle_spider_completion():
+    """
+    Handle actions to perform when the spider completes crawling.
+    """
+    global is_scraping
+    is_scraping = False
+    logger.info("Scrapy crawl completed and flag reset.")
```

Comment:
`nit`: might be helpful to log the current value of flag

Reply:
Got it.